python/ray/train/constants.py

```diff
@@ -99,3 +108,12 @@ def _get_defaults_results_dir() -> str:
 
 # Key for AIR Checkpoint world rank in TrainingResult metadata
 CHECKPOINT_RANK_KEY = "checkpoint_rank"
+
+# Map of supported accelerators to the environment variable configuration
+# 1/sharing cores/devices across workers 2/runtime environment variable to configure.
+SUPPORTED_ACCELERATOR_DEVICES_TO_CONFIG = {
+    ray_constants.NEURON_CORES: (
+        ENABLE_SHARE_NEURON_CORES_ACCELERATOR_ENV,
+        ray_constants.NEURON_RT_VISIBLE_CORES_ENV_VAR,
+    )
+}
```

Comment:
Can we convert this to a dataclass? Primarily to make the typing more clear than an arbitrary tuple.

```python
@dataclass
class ResourceConfig:
    """ 


    """
    resource_name: str
    resource_ids_env_var: str
    share_resource_ids_env_var: str


RESOURCE_CONFIGS = [ResourceConfig(ray_constants.NEURON_CORES,
        ray_constants.NEURON_RT_VISIBLE_CORES_ENV_VAR, ENABLE_SHARE_NEURON_CORES_ACCELERATOR_ENV)]
SUPPORTED_RESOURCES_TO_CONFIG = {resource_config.resource_name: resource_config for resource_config in RESOURCE_CONFIGS}

Reply:
Good refactor, added.